packages/trader/src/AppV2/Components/TradeParameters/Duration/day.tsx

```diff
@@ -53,33 +56,78 @@ const DayInput = ({
         duration_min_max,
         trade_types,
         contract_type,
+        duration,
+        tick_data,
+        symbol,
+        barrier_1,
     } = useTraderStore();
     const trade_store = useTraderStore();
+    const { addSnackbar } = useSnackbar();
 
     const new_values = {
         duration_unit: 'd',
-        duration: day || '',
-        expiry_time: null,
+        duration: day || duration,
         expiry_type: 'duration',
         contract_type,
+        basis: 'stake',
+        amount: '5',
+        symbol,
     };
+
     const proposal_req = getProposalRequestObject({
         new_values,
         trade_store,
         trade_type: Object.keys(trade_types)[0],
     });
 
-    const { data: response } = useDtraderQuery<ProposalResponse>(['proposal', JSON.stringify(day)], proposal_req, {
-        enabled: day !== null,
-    });
+    const { data: response } = useDtraderQuery<ProposalResponse>(
+        ['proposal', JSON.stringify(day)],
+        (() => {
+            const request_payload = {
+                ...proposal_req,
+                ...{ symbol },
+            };
+
+            if (barrier_1) {
+                return {
+                    ...request_payload,
+                    barrier: Math.round(tick_data?.quote as number),
+                };
+            }
+            return request_payload;
+        })(),
+        {
+            enabled: trigger_date,
+        }
+    );
 
     useEffect(() => {
-        if (response?.proposal?.date_expiry) {
-            setTempExpiryTime(
-                new Date((response?.proposal?.date_expiry as number) * 1000).toISOString().split('T')[1].substring(0, 8)
-            );
+        if (response) {
+            if (response?.error?.message && response?.error?.details?.field === 'duration') {
+                addSnackbar({
+                    message: <Localize i18n_default_text={response?.error?.message} />,
+                    status: 'fail',
+                    hasCloseButton: true,
+                    style: { marginBottom: '48px' },
+                });
+                setIsDisabled(true);
+            } else {
+                setIsDisabled(false);
+            }
+
+            if (response?.proposal?.date_expiry) {
+                setExpiryTimeInput(
+                    new Date((response?.proposal?.date_expiry as number) * 1000)
+                        .toISOString()
+                        .split('T')[1]
+                        .substring(0, 8)
```

Comment:
`.split('T')[1].substring(0, 8`) doesn't seems to be super safe, but I can't suggest something

Reply:
I mean you can ignore my comment

Reply:
I think it won't be an unsafe unless the response?.proposal?.date_expiry is something invalid like undefined . And so I already added check to see if response?.proposal?.date_expiry is valid.